serving-api/src/main/scala/com/stratio/sparkta/serving/api/actor/ClusterLauncherActor.scala

```diff
@@ -0,0 +1,229 @@
+                     driverParams: Seq[String]): Unit = {
+    val sparkLauncher = new SparkLauncher()
+      .setSparkHome(sparkHome)
+      .setAppResource(hdfsDriverFile)
+      .setMainClass(main)
+      .setMaster(master)
+    args.map({ case (k: String, v: String) => sparkLauncher.addSparkArg(k, v) })
+    if (isStandaloneSupervise) sparkLauncher.addSparkArg(StandaloneSupervise)
+    //Spark params (everything starting with spark.)
+    sparkConf.map({ case (key: String, value: String) => sparkLauncher.setConf(key, value) })
+    // Driver (Sparkta) params
+    driverParams.map(sparkLauncher.addAppArgs(_))
+    val spark = sparkLauncher.launch()
+    spark.waitFor()
+    val exit = spark.exitValue()
+    log.debug(s"Spark exit status: $exit")
+    // TODO: This should be done in another thread. At this point, you only see the error when job is terminated
+    val in = Source.fromInputStream(spark.getErrorStream)
+    in.getLines().foreach(log.error)
+    in.close()
+    if (exit != 0) setErrorStatus
+  }
+
+  private def validatePolicyJars(activeJars: Either[Seq[String], Seq[String]]): Unit = {
+    if (activeJars.isLeft) {
+      activeJars.left.get.foreach(log.error)
+      require(activeJars.isRight, s"The policy have jars witch cannot be found in classpath:")
+    }
+    require(activeJars.right.get.nonEmpty, s"The policy don't have jars in the plugins")
+  }
+
+  private def sparkArgs: Map[String, String] =
+    ClusterLauncherActor.toMap(AppConstant.DeployMode, "--deploy-mode", clusterConfig) ++
+      ClusterLauncherActor.toMap(AppConstant.NumExecutors, "--num-executors", clusterConfig) ++
+      ClusterLauncherActor.toMap(AppConstant.ExecutorCores, "--executor-cores", clusterConfig) ++
+      ClusterLauncherActor.toMap(AppConstant.TotalExecutorCores, "--total-executor-cores", clusterConfig) ++
+      ClusterLauncherActor.toMap(AppConstant.ExecutorMemory, "--executor-memory", clusterConfig) ++
+      // Yarn only
+      ClusterLauncherActor.toMap(AppConstant.YarnQueue, "--queue", clusterConfig)
+
+  private def render(config: Config, key: String): String = config.atKey(key).root.render(ConfigRenderOptions.concise)
+
+  private def encode(value: String): String = BaseEncoding.base64().encode(value.getBytes)
+
+  private def zkConfigEncoded: String = encode(render(zookeeperConfig, "zookeeper"))
+
+  private def detailConfigEncoded: String = encode(render(detailConfig, "config"))
+
+  private def sparkConf: Seq[(String, String)] = clusterConfig.entrySet()
+    .filter(_.getKey.startsWith("spark."))
+    .toSeq
+    .map(e => (e.getKey, e.getValue.toString))
+}
+
+object ClusterLauncherActor extends SLF4JLogging {
+
+  def toMap(key: String, newKey: String, config: Config): Map[String, String] =
+    Try(config.getString(key)) match {
+      case Success(value) => Map(newKey -> value)
+      case Failure(_) => {
+        log.debug(s"The key $key was not defined in config.")
+        Map.empty[String, String]
+      }
+    }
+}
```

Comment:
IMHO this class is too long, maybe you can extract something to other class. BTW  Good Job!!